plugins/src/ios/OptimoveSDKPlugin.swift

```diff
@@ -1,14 +1,51 @@
-@objc(initialize:)
-  func initialize(command: CDVInvokedUrlCommand) {
-    let args = command.arguments
-    
-    let inAppConsentStrategy: InAppConsentStrategy = {
-      if let strategy = args?[2] as? String {
-        return .init(rawValue: strategy)!
-      }
-      return.notEnabled
-    }()
-
-    let config = OptimoveConfigBuilder(optimoveCredentials: args?[0] as? String, optimoveCredentials: args?[1] as? String).enableInAppMessaging(inAppConsentStrategy: inAppConsentStrategy).build()
-    Optimove.initialize(with: config)
-  }
\ No newline at end of file
+import OptimoveSDK
+import NotificationCenter
+@objc(Optimobile_Cordova) class OptimobileSDKPlugin : CDVPlugin {
+    private let internalQueue = DispatchQueue(label: "com.singletioninternal.queue",
+                                              qos: .default,
+                                              attributes: .concurrent)
+    
+    private var pendingPush: PushNotification?
+    
+    private let optimoveCredentialsKey = "optimoveCredentials"
+    private let optimoveMobileCredentialsKey = "optimoveMobileCredentials"
+    
+    private let credentialsJsonName = "optimove"
+    
+    func load() {
+        internalQueue.sync(flags: .barrier) {
+            NotificationCenter.default.addObserver(self, selector: #selector(self.didFinishLaunching(notification:)), name: UIApplication.didFinishLaunchingNotification, object: nil)
+        }
+    }
+    
+    @objc func didFinishLaunching(notification: NSNotification) {
+        let configPath = Bundle.main.path(forResource: "optimove", ofType: "plist")
+        
+        guard let configPath = configPath else {
+            print("optimove.plist NOT FOUND");
+            return
+        }
+        
+        guard let configValues: [String: String] = NSDictionary(contentsOfFile: configPath) as? [String: String] else {
+            print("optimove.plist IS NOT VALID");return
+        }
+        
+        if let userInfo = notification.userInfo {
+            let userInfoDict = notification.userInfo?[UIApplication.LaunchOptionsKey.remoteNotification];
+            if userInfoDict != nil {
+                pendingPush = PushNotification.init(userInfo: userInfo, response: nil)
+            }
+        }
+        
+        let config = OptimoveConfigBuilder(optimoveCredentials: configValues[optimoveCredentialsKey], optimobileCredentials: configValues[optimoveMobileCredentialsKey])
+        
+        if configValues["inAppConsentStrategy"] == "auto-enroll" {
+            config.enableInAppMessaging(inAppConsentStrategy: .autoEnroll)
+        }
+        else if configValues["inAppConsentStrategy"] == "explicit-by-user" {
+            config.enableInAppMessaging(inAppConsentStrategy: .explicitByUser)
+        }
```

Comment:
Can we defer these pieces to the messaging branches?

Reply:
sure i have one open so when i make a pr from it will check that piece of code